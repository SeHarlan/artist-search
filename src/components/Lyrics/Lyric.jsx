import React from 'react'; 
import { useFetchLyrics } from '../../services/hooks';
import { useParams } from 'react-router-dom';
import style from './lyric.css';

const Lyric = () => {
  const { artist, title } = useParams();

  const body = useFetchLyrics(artist, title);
  
  return (
    <section className={style.lyric}>
      <h2>{title}</h2>
      <p>{body ? body : 'Sorry, No Lyrics Found'}</p>
    </section>
  );
};


export default Lyric;
