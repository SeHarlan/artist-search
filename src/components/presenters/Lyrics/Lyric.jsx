import React from 'react'; 
import PropTypes from 'prop-types';
import { useFetchLyrics } from '../../../services/hooks';

const Lyric = ({ match }) => {
  const artist = match.params.artist;
  const title = match.params.title;

  const body = useFetchLyrics(artist, title);
  
  return (
    <>
      <h2>{title}</h2>
      <p>{body}</p>
    </>
  );
};

Lyric.propTypes = {
  match: PropTypes.object.isRequired
};

export default Lyric;
