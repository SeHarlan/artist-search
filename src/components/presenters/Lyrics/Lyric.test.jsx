import React from 'react';
import { shallow } from 'enzyme';
import Lyric from './Lyric';

describe('Lyric component', () => {
  it('renders Lyric', () => {
    const wrapper = shallow(<Lyric title="test" body="lyrics" />);
    expect(wrapper).toMatchSnapshot();
  });
});
