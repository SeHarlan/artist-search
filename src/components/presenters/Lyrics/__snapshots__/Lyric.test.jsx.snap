// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lyric component renders Lyric 1`] = `
<Fragment>
  <h2>
    test
  </h2>
  <p>
    lyrics
  </p>
</Fragment>
`;
